pkg/model/keycloak_metrics_route.go

```diff
@@ -0,0 +1,56 @@
+package model
+
+import (
+	kc "github.com/keycloak/keycloak-operator/pkg/apis/keycloak/v1alpha1"
+	v1 "github.com/openshift/api/route/v1"
+	v12 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func KeycloakMetricsRoute(cr *kc.Keycloak, keycloakMainRoute *v1.Route) *v1.Route {
+	keycloakMainRouteCopy := keycloakMainRoute.DeepCopy()
+	return &v1.Route{
+		ObjectMeta: v12.ObjectMeta{
+			Name:      KeycloakMetricsRouteName,
+			Namespace: cr.Namespace,
+			Labels: map[string]string{
+				"app": ApplicationName,
+			},
+			Annotations: map[string]string{
+				"haproxy.router.openshift.io/balance":        RouteLoadBalancingStrategy,
+				"haproxy.router.openshift.io/rewrite-target": KeycloakMetricsRouteRewritePath,
+			},
+		},
+		Spec: v1.RouteSpec{
+			Host: keycloakMainRouteCopy.Spec.Host,
+			Path: KeycloakMetricsRoutePath,
+			Port: keycloakMainRouteCopy.Spec.Port,
+			TLS:  keycloakMainRouteCopy.Spec.TLS,
+			To:   keycloakMainRouteCopy.Spec.To,
+		},
+	}
+}
+
+func KeycloakMetricsRouteReconciled(cr *kc.Keycloak, currentState *v1.Route, keycloakMainRoute *v1.Route) *v1.Route {
```

Comment:
The `cr` argument is unused here but I guess it's to keep it consistent with other reconciler models.

Reply:
Yes, exactly. I like keeping them consistent. Oftentimes you copy-paste stuff around in the model package.